### ISSUE TITLE
Failing to build MPI application, missing MPI runner

### ISSUE THREAD

**Issue by Carltoffel** [fortran-lang/fpm#935]
### Description

I want to build an MPI application, but there is no MPI runner on the login node because you have to submit a sbatch job.
I can `fpm test` my program (and I assume the same goes for `fpm run`) if I replace [this function call](https://github.com/fortran-lang/fpm/blob/6d33e74c410049d2ab9df19ccb6e8135c03b9a59/src/fpm_meta.f90#L1008) with my sbatch runner.
But when using `fpm build`, a runner shouldn't be necessary. `fpm build` terminates before it starts compiling anything with the following error message:
```
<ERROR> *cmd_build* Model error: cannot find a valid mpi runner command
```

### Expected Behaviour

`fpm build` shouldn't require an MPI runner in the path.

### Version of fpm

0.9.0, 6d33e74c410049d2ab9df19ccb6e8135c03b9a59

### Platform and Architecture

Rocky Linux 8.7 (Green Obsidian)

### Additional Information

GNU Fortran (GCC) 11.3.0
OpenMPI 4.1.4

**Edit**: I did some changes to `fpm` which affected the behaviour I described. Therefore, I fixed the description of symptoms.

**Comment by Carltoffel** [comment]
After some more testing I realized, that `fpm test` (and probably `fpm run`, too) don't work, even if I provide the runner with the `--runner` option.

**Comment by perazz** [comment]
Solved by #937
